mmv1/products/netapp/storagePool.yaml

```diff
@@ -0,0 +1,130 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'storagePool'
+description: |
+  StoragePool is a container for volumes with a service level and capacity.
+  Volumes can be created in a pool of sufficient available capacity. StoragePool capacity is what you are billed for.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'QUICKSTART_TITLE': 'https://cloud.google.com/netapp/volumes/docs/reference/rest/v1/projects.locations.storagePools'
+  api: 'https://cloud.google.com/netapp/volumes/docs/reference/rest/v1/projects.locations.storagePools'
+base_url: projects/{{project}}/locations/{{location}}/storagePools
+self_link: projects/{{project}}/locations/{{location}}/storagePools/{{storage_pool_id}}
+create_url: projects/{{project}}/locations/{{location}}/storagePools?storagePoolId={{storage_pool_id}}
+update_url: projects/{{project}}/locations/{{location}}/storagePools/{{storage_pool_id}}
+update_verb: :PATCH
+update_mask: true
+delete_url: projects/{{project}}/locations/{{location}}/storagePools/{{storage_pool_id}}
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+id_format: 'projects/{{project}}/locations/{{location}}/storagePools/{{storage_pool_id}}'
+import_format: ['projects/{{project}}/locations/{{location}}/storagePools/{{storage_pool_id}}']
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      LOCATION_DESCRIPTION
+  - !ruby/object:Api::Type::String
+    name: 'storage_pool_id'
+    description:
+      'The unique name of the storage pool
+      `[_a-zA-Z0-9][-_.a-zA-Z0-9]*`.'
+    required: true
+    immutable: true
+    url_param_only: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'Storage_pool_create'
+    primary_resource_id: 'test_pool'
+    vars:
+      pool_name: 'test-pool'
+      network_name: 'test-network'
+      global_name: 'test-address'
+properties:
+  # Fields go here
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: |
+      Name of the storage pool
+  - !ruby/object:Api::Type::Enum
+    name: 'serviceLevel'
```

Comment:
looks like this marked as `required` in [the API doc](https://cloud.google.com/netapp/volumes/docs/reference/rest/v1/projects.locations.storagePools). Is this field required? If so, let's add `required: true`